.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,42 @@
+name: Docker Build and Publish
+
+on:
+  workflow_dispatch:
+    inputs:
+      tagName:
+        description: "Tag name"
+        required: true
+        default: 'v1'
+        
+jobs:
+  push_to_registry:
+    runs-on: ubuntu-latest
+
+    name: Build docker image and push to dockerhub
+
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3 
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
```

Comment:
you would need to add these as action secrets in the repo settings

Reply:
Even the free account on docker hub is sufficient btw. You don't need to buy any subscription there.

Reply:
How about using the GitHub container registry instead? It seems to work pretty seamlessly with e.g. `docker-compose`

Reply:
Aio needs the images to be published on docker hub bwcause update chwcks do not work with github container registry. However you could of course publish it on both

Reply:
Ok I see. Let me set that up then

Reply:
I guess you only need to create a free account, add a secret in the account settings and add both account and secret to this repo as variable for actions. Then running the workflow should suceed and automatically create the repo on docker hub